dotnet/src/Connectors/Connectors.Memory.InMemory/InMemoryVectorStoreRecordCollection.cs

```diff
@@ -113,13 +113,19 @@ public Task<bool> CollectionExistsAsync(CancellationToken cancellationToken = de
     /// <inheritdoc />
     public Task CreateCollectionAsync(CancellationToken cancellationToken = default)
     {
-        if (!this._internalCollections.ContainsKey(this._collectionName))
+        if (!this._internalCollections.ContainsKey(this._collectionName)
```

Comment:
Can remove the ContainsKey, given that TryAdd is being used below, no?

Reply:
Yes, but then we would be always allocating a dictionary first.